activerecord/lib/active_record/relation/batches.rb

```diff
@@ -222,6 +222,8 @@ def in_batches(of: 1000, start: nil, finish: nil, load: false, error_on_ignore:
       end
 
       batch_limit = of
+      raise ArgumentError, "batch_limit cannot be passed as nil" if batch_limit.nil?
```

Comment:
I think you should move this check higher - before the check for `order`, for example. And change the message to something like `:of should not be nil`, because user passed `:of`, not `:batch_order`.

Reply:
changed message 👍 
`:of should not be nil`
https://github.com/rails/rails/pull/45733/commits/35f5b20f415f0e85842ea9321f8f01c29a595d6b#diff-8facabeda611c44c59470e0f632a5805f152e40ec63ed7a75c76896e783ddb5dR210